posthog/api/test/test_kafka_inspector.py

```diff
@@ -0,0 +1,51 @@
+import json
+from typing import Dict, List, Union
+from unittest.mock import patch
+
+from rest_framework import status
+
+from posthog.api.kafka_inspector import KafkaConsumerRecord
+from posthog.test.base import APIBaseTest
+
+
+class TestKafkaInspector(APIBaseTest):
+    def setUp(self):
+        super().setUp()
+        self.user.is_staff = True
+        self.user.save()
+
+    def _to_json(self, data: Union[Dict, List]) -> str:
+        return json.dumps(data)
+
+    @patch(
+        "posthog.api.kafka_inspector.get_kafka_message",
+        side_effect=lambda _, __, ___: KafkaConsumerRecord("foo", 0, 0, 1650375470233, "k", "v"),
+    )
+    def test_fetch_message(self, _):
+        response = self.client.post(
```

Comment:
maybe try to get this twice just to test that we can (i.e. that we reset offsets properly)?

Reply:
so turns out we don't have a great story for testing the producer actually working so I'm mocking it here.

I did manually test this and you can see in the video that we can go back.

Also note this a different consumer group so it won't ever affect anything.

I'll punt but definitely look closely into e2e Kafka tests if we release this more broadly to others